n3fit/src/n3fit/layers/DY.py

```diff
@@ -19,48 +32,35 @@ def gen_mask(self, basis):
                 basis_mask[i, j] = True
         return op.numpy_to_tensor(basis_mask, dtype=bool)
 
-    def call(self, pdf_raw):
-        """
-        This function perform the fktable \otimes pdf \otimes pdf convolution.
+    def mask_fk(self, fk, mask):
+        if self.num_replicas > 1:
+            return op.einsum('fgF, nFxy -> nxfyg', mask, fk)
+        else:
+            mask = op.einsum('fgF -> Ffg', mask)
+            fk = op.einsum('nFxy -> nFyx', fk)
+            mask_and_fk = (mask, fk)
+            return mask_and_fk
```

Comment:
```suggestion
        mask = op.einsum('fgF -> Ffg', mask)
        fk = op.einsum('nFxy -> nFyx', fk)
        return mask, fk
```

Please add a docstr and a comment on why this difference between multireplicas and single replicas is necessary.

Reply:
Done!